### PR TITLE
Fix Delete repository message

### PR DESCRIPTION
**Pull request by muhfaris**
Fix issue [#3201](https://github.com/go-gitea/gitea/issues/3201).
Get name of repository in message before delete repository

> no. <strong>%s</strong> should be replaced but in fact didn't

-  ~~Create new function to get repo name~~
-  ~~Create Routing repo by id~~
-  ~~Ajax - function takes the repo name~~
-  ~~Change message from newline tags `(<br>)` with paragraph tags `(<p>)`~~

- [x] Add tag `data-repo-name`
- [x] Add repository name to the message delete repository

scrot_2017-12-25--1514165487_764x361

### CODE REVIEW

File: public/js/index.js
Comment (strk) [inline public/js/index.js]: This pattern seems too generic for an URL, what it hostname itself contains a "del" substring ?

File: routers/routes/routes.go
Comment (strk) [inline routers/routes/routes.go]: Why Post, for a GetSomething?

**Review by strk — CHANGES_REQUESTED** [review]

File: routers/routes/routes.go
Comment (muhfaris) [reply routers/routes/routes.go]: yes, i create new route in `routes.go` and function in `repos.go`. that code use to get repository name by id. the result, repository name will be display in delete repository message

File: public/js/index.js
Comment (muhfaris) [reply public/js/index.js]: the pattern use to get uri from `/admin/repos/page?delete={id_repo}` to   `/admin/repos/`

File: templates/admin/repo/list.tmpl
Comment (ethantkoenig) [inline templates/admin/repo/list.tmpl]: I think that the right solution is to simply pass a format argument to `.i18n.Tr "repo.settings.delete_notices_2"` (see [user/dashboard/navbar.tmpl](https://github.com/go-gitea/gitea/blob/f5155b99136b6e19ab494a97adfcee1810a3d5e7/templates/user/dashboard/navbar.tmpl#L45) for an example of what I have in mind)

This would be a one-line change, and would not require new routes or any client-side changes.

**Review by ethantkoenig — CHANGES_REQUESTED** [review]

File: templates/admin/repo/list.tmpl
Comment (muhfaris) [reply templates/admin/repo/list.tmpl]: yes, i think so. but after check in the `repos.go` is imposible, cause the function `deleterepo` will be call after approve button (yes). so, it is imposibble. the issue need name repository when it is deleted.

Comment (lafriks) [reply templates/admin/repo/list.tmpl]: You can add `data-repo-name` to link just like `data-id` is added and use $(this).data('repo-name') in on click javascript to get repo name

Comment (muhfaris) [reply templates/admin/repo/list.tmpl]: okay, i will update and update the code

**Comment by bkcsoft** [comment]
@muhfaris could you update the description to reflect the changes done? I don't see a reference in the changes to these:
```
- [x] Create new function to get repo name
- [x] Create Routing repo by id
```

File: public/js/index.js
Comment (bkcsoft) [inline public/js/index.js]: Could you explain what this is supposed to do? Since I think this does way more than you hope it will 🤔

Comment (muhfaris) [reply public/js/index.js]: i use regex to get string `<strong>%s</strong>`  from the message delete repository. the message as follows.
`- This operation will permanently delete everything in the <strong>%s</strong> repository, including code, issues, comments, the wiki, and collaborators associations.`

then, will be replace with `<b>{repository_name}</b>` and it will be as follows
`- This operation will permanently delete everything in the <b>{repository_name}</b> repository, including code, issues, comments, the wiki, and collaborators associations.`

because  `<strong>` tag is not work i change with `<b>` tag, I think not a bad idea

Comment (lafriks) [reply public/js/index.js]: I think it should just replace `%s` and `%[1]s` without using additional regexp

Comment (muhfaris) [reply public/js/index.js]: i tried it but not work

scrot_2017-12-27--1514365900_721x235

Comment (lafriks) [reply public/js/index.js]: `strong` will work, you need to add safe html when generating this dialog in golang template. Best option would be to add "{% repo_name %}" to $.i18n.Tr as second argument and than replace it

Comment (ethantkoenig) [reply public/js/index.js]: If we want to use `<b>` instead of `<strong>`, we should just change the locale, instead of introducing client-side logic to change `<strong>` tags to `<b>`.

Also, won't this regexp match `<b>...</b>`, not `<strong>...</strong>`?

Comment (muhfaris) [reply public/js/index.js]: @lafriks sorr, I don't understand what you mean.

Comment (muhfaris) [reply public/js/index.js]: @ethantkoenig  i delete the regex and change from `b` tag to `strong`

File: options/locale/locale_en-US.ini
Comment (ethantkoenig) [inline options/locale/locale_en-US.ini]: This locale is used in multiple places, so I don't think removing the `<strong>` tags is the right thing to do.

Apologies if my previous comment was not clear, but if we now plan to continue rendering with `<strong>` tags, changing the locale is not necessary.

Comment (ethantkoenig) [reply options/locale/locale_en-US.ini]: Also, if we add the `<strong>` tags back to the locale, maybe it'd be best to add a class (e.g. `<strong class="repo-name">%s</string>`) so we can just select/modify/remove the repo name using JQuery (which would make the Javascript a little cleaner).

**Review by ethantkoenig — CHANGES_REQUESTED** [review]

File: options/locale/locale_en-US.ini
Comment (strk) [reply options/locale/locale_en-US.ini]: agreed, keep the `<strong>` in locale, and only expect to find `%s` when you replace later

Comment (muhfaris) [reply options/locale/locale_en-US.ini]: okay, i reverted the strong tag

**Comment by strk** [comment]
Please rebase to master and force-push (less commits, cleaner merge, easier to review).

I don't really understand what's going on with all that HTML encoding: why do you need to replace the escaped HTML to unescaped one? What if a translation doesn't include `<strong>` in the template ? I'd feel better with a generic solution to html escaping while only replacing `%s` ... is that possible ?

**Comment by lunny** [comment]
LGTM

**Comment by muhfaris** [comment]
@strk yes, sorry my mistake. i think so and i will be update the code

**Comment by strk** [comment]
LGTM

**Comment by lunny** [comment]
@strk you should approval

**Review by strk — APPROVED** [review]

File: public/js/index.js
Comment (thehowl) [inline public/js/index.js]: just to be safe, i'd prefer if either .text() was called instead of html to set the content, or if repo-name was run through an html escaper

**Review by thehowl — CHANGES_REQUESTED** [review]

File: public/js/index.js
Comment (muhfaris) [reply public/js/index.js]: if use `text()`, `strong` tag not work
scrot_2018-01-09--1515502298_667x59

Comment (lafriks) [reply public/js/index.js]: Because that is problem how it is rendered in template , it must have filter safe html set

**Review by thehowl — APPROVED** [review]
I would say to sanitize the input first, but seeing as repository names don't allow \<\>, LGTM

File: public/js/index.js
Comment (thehowl) [reply public/js/index.js]: No, that's because of text which escapes the strong tag

File: public/js/index.js
Comment (lafriks) [inline public/js/index.js]: What if repo name will match any word in that message?

**Review by lafriks — CHANGES_REQUESTED** [review]

File: public/js/index.js
Comment (muhfaris) [reply public/js/index.js]: if you look into the code, first time symbol `%s` will be replace  with repository name `$this.data('repo-name')` and then if cancel to remove repository the repository name will be replace with `%s`

Comment (lafriks) [reply public/js/index.js]: But what is if repository name will be `operation`? ;)

Comment (lafriks) [reply public/js/index.js]: Just save message contents before replacing in variable and restore it back later on

**Comment by strk** [comment]
> Just save message contents before replacing in variable and restore it back later on

+1

File: public/js/index.js
Comment (muhfaris) [reply public/js/index.js]: @lafriks sorry i'm not understand with your question ( But what is if repository name will be operation?) 

i think not necessary to save message content in variabel, because is same. the content message after deleted repository is will be reset to default and tested after deleted repository or cancel delete repository, the message content is works

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
@ethantkoenig need your approval

**Review by appleboy — APPROVED** [review]

**Review by ethantkoenig — APPROVED** [review]

**Pull request merged**
